internal/nonkube/client/fs/path_provider.go

```diff
@@ -1,19 +1,17 @@
 package fs
 
-import "fmt"
+import (
+	"github.com/skupperproject/skupper/pkg/nonkube/api"
+)
 
 type PathProvider struct {
 	Namespace string
 }
 
-func (p *PathProvider) getDefaultNamespace() string {
-	return ".local/share/skupper/namespaces/default/input/sources"
-}
-
 func (p *PathProvider) GetNamespace() string {
+	return api.GetHostNamespaceHome(p.Namespace) + "/input/sources"
+}
 
-	if p.Namespace == "" {
-		return p.getDefaultNamespace()
-	}
-	return fmt.Sprintf(".local/share/skupper/namespaces/%s/input/sources", p.Namespace)
+func (p *PathProvider) GetRuntimeNamespace() string {
+	return api.GetHostNamespaceHome(p.Namespace) + "/runtime/state"
```

Comment:
suggest replace "/runtime/state" with api.RuntimeSiteStatePath and concatenate with path.Join()

Reply:
fixed